docs/src/components/hardware-list.tsx

```diff
@@ -0,0 +1,160 @@
+  HardwareMetadata["type"],
+  Record<"singular" | "plural", string>
+> = {
+  board: { plural: "Boards: ", singular: "Board: " },
+  shield: { singular: "Shield: ", plural: "Shields: " },
+  interconnect: { singular: "Interconnect: ", plural: "Interconnects: " },
+};
+
+function HardwareLineItem({ item }: { item: HardwareMetadata }) {
+  return (
+    <li>
+      <a href={item.url}>{item.name}</a> (
+      {TYPE_LABELS[item.type][itemHasMultiple(item) ? "plural" : "singular"]}{" "}
+      {itemIds(item)})
+    </li>
+  );
+}
+
+interface InterconnectDetails {
+  interconnect?: Interconnect;
+  boards: Array<Board>;
+  shields: Array<Shield>;
+}
+
+function mapInterconnect({
+  interconnect,
+  boards,
+  shields,
+}: InterconnectDetails) {
+  if (!interconnect) {
+    return null;
+  }
+
+  return (
+    <div key={interconnect.id}>
+      <h4>{interconnect.name} Keyboards</h4>
+      {interconnect.description && <p>{interconnect.description}</p>}
+      <h5>Boards</h5>
+      <ul>
+        {boards.map((s) => (
+          <HardwareLineItem key={s.id} item={s} />
+        ))}
+      </ul>
+      <h5>Shields</h5>
+      <ul>
+        {shields.map((s) => (
+          <HardwareLineItem key={s.id} item={s} />
+        ))}
+      </ul>
+    </div>
+  );
+}
+
+function HardwareList({ items }: HardwareListProps) {
+  let grouped: {
+    onboard: Array<Board>;
+    interconnects: Record<string, InterconnectDetails>;
+  } = items.reduce(
+    (agg, hm) => {
+      switch (hm.type) {
+        case "board":
+          if (hm.features?.includes("keys")) {
+            agg.onboard.push(hm);
+          } else if (hm.exposes) {
+            hm.exposes.forEach((element) => {
+              let ic = agg.interconnects[element] || {
+                boards: [],
+                shields: [],
+              };
+              ic.boards.push(hm);
+              agg.interconnects[element] = ic;
+            });
+          } else {
+            console.error("Board without keys or interconnect");
+          }
+          break;
+        case "shield":
+          hm.requires.forEach((id) => {
+            let ic = agg.interconnects[id] || { boards: [], shields: [] };
+            ic.shields.push(hm);
+            agg.interconnects[id] = ic;
+          });
+          break;
+        case "interconnect":
+          let ic = agg.interconnects[hm.id] || { boards: [], shields: [] };
+          ic.interconnect = hm;
+          agg.interconnects[hm.id] = ic;
+          break;
+      }
+      return agg;
+    },
+    { onboard: [] as Array<Board>, interconnects: {} }
+  );
+
+  return (
+    <>
+      <h2>Keyboards</h2>
+      <h3>Onboard Controller Boards</h3>
+      <p>
+        Keyboards with onboard controllers are single PCBs that contain all the
+        components of a keyboard, including the controller chip, switch
+        footprints, etc.
+      </p>
+      <ul>
+        {grouped["onboard"]
+          .sort((a, b) => a.name.localeCompare(b.name))
```

Comment:
If we ever start getting into things with double digit version numbers, we might want to pull in something like https://www.npmjs.com/package/string-natural-compare to sort numbers embedded in strings numerically instead of lexicographically.